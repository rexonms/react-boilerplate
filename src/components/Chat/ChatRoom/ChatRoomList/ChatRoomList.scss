.container {
  //height: 400px;
}
.autoScrollContainer {
  overflow: scroll;
}
.content {
  display: flex;
  width: 100%;
  margin-bottom: 10px;
}
.side {
  width: 70%;
}
.user {
  justify-content: flex-end;
}
.friend {
  justify-content: flex-start;
}