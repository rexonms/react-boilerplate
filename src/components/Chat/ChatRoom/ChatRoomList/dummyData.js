// http://blog.slatepeak.com/creating-a-real-time-chat-api-with-node-express-socket-io-and-mongodb/

export const CHAT_USER_EMAIL_ADDRESS = 'rexonms@gmail.com';
export const CHAT_FRIEND_EMAIL_ADDRESS = 'john@mail.com';
// List of all the conversations
// All the conversations
export const conversationList = [
  {
    id: '1',
    authorEmail: CHAT_USER_EMAIL_ADDRESS,
    message: 'hola!',
    imageURL: 'https://place-hold.it/100',
    imageAltText: 'Alt text of image',
  },
  {
    id: '2',
    authorEmail: CHAT_FRIEND_EMAIL_ADDRESS,
    message: 'whats up!',
    imageURL: 'https://place-hold.it/100',
    imageAltText: 'Alt text of image',
  },
  {
    id: '3',
    authorEmail: CHAT_FRIEND_EMAIL_ADDRESS,
    message: 'Where did you go? Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: ',
    imageURL: 'https://place-hold.it/100',
    imageAltText: 'Alt text of image',
  },
  {
    id: '4',
    authorEmail: CHAT_USER_EMAIL_ADDRESS,
    message: 'Oops sorry! didn\'t se you message',
    imageURL: 'https://place-hold.it/100',
    imageAltText: 'Alt text of image',
  },
  {
    id: '5',
    authorEmail: CHAT_USER_EMAIL_ADDRESS,
    message: 'In addition, you should probably review the links in the \'See Also\' section of that document. I especially recommend the video presentations.\n' +
    '\n' +
    'Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: ',
    imageURL: 'https://place-hold.it/100',
    imageAltText: 'Alt text of image',
  },
];
