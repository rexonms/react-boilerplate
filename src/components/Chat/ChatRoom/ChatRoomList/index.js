import React from 'react';
import PropTypes from 'prop-types';

import Styles from './ChatRoomList.scss';
import ChatRoomItem from '../ChatRoomItem/index';
import { CHAT_BOX_USER, CHAT_BOX_FRIEND } from '../ChatRoomItem/Const';

const propTypes = {
  /** Name of the component */
  list: PropTypes.shape({
    /** The conversation id */
    id: PropTypes.string,
    /** Email address of the author */
    authorEmail: PropTypes.string,
    /** Message sent by user */
    message: PropTypes.string,
  }).isRequired,
};

const defaultProps = {};
const userEmail = 'rexonms@gmail.com';

const ChatRoomList = props => (
  <div className={`${Styles.container}`}>
    {Object.keys(props.list).map((item) => {
      const conversation = props.list[item];
      const type = userEmail === conversation.authorEmail ? CHAT_BOX_USER : CHAT_BOX_FRIEND;
      return (
        <div key={`conv-${conversation.id}`} className={`${Styles.content} ${Styles[type]}`}>
          <ChatRoomItem
            id={conversation.id}
            message={conversation.message}
            type={type}
            imageURL={conversation.imageURL}
            imageAltText={conversation.imageAltText}
          />
        </div>
      );
    })}
  </div>
);

ChatRoomList.defaultProps = defaultProps;
ChatRoomList.propTypes = propTypes;

export default ChatRoomList;
