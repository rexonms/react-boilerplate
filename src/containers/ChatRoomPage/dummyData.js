const chatterAEmail = 'rexonms@gmail.com';
const chatterBEmail = 'john@mail.com';
export const chatterAImage = 'http://dummyimage.com/mediumrectangle/111111/eeeeee';
export const chatterBImage = 'http://dummyimage.com/mediumrectangle/cccccc/eeeeee';
export const dummyMessages = {
  conversations: {
    1: {
      id: '1',
      chatterAEmail,
      chatterBEmail,
      messages: [
        {
          id: '1',
          message: 'Hello!',
          authorEmail: chatterAEmail,
          recipientEmail: chatterBEmail,
        },
        {
          id: '2',
          message: 'WhatS up!',
          authorEmail: chatterBEmail,
          recipientEmail: chatterAEmail,
        },
        {
          id: '3',
          message: 'How is life?',
          authorEmail: chatterBEmail,
          recipientEmail: chatterAEmail,
        },
        {
          id: '4',
          message: 'Do you know? There are two options, either you should spy on _handleCloneClick of component\'s prototype, before you render the component.',
          authorEmail: chatterBEmail,
          recipientEmail: chatterAEmail,
        },
      ],
    },
  },
};
