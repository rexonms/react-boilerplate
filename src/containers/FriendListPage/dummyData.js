const friendListPageData = {
  translations: {
    status: {
      online: 'Online',
      offline: 'Offline',
    },
  },
  dbData: {
    friendsList: [
      {
        id: '1',
        conversationId: '100',
        avatarImageURL: 'http://dummyimage.com/mediumrectangle/111111/eeeeee',
        recipientName: 'John Smith',
        lastMessage: 'Where have you been?',
        recipientEmail: 'john@mail.com',
      },
      {
        id: '2',
        conversationId: '101',
        avatarImageURL: 'http://dummyimage.com/mediumrectangle/333333/eeeeee',
        recipientName: 'Jenny',
        lastMessage: 'And then there this time when I was thing of lots of stuff. A single line field is used when the input anticipated by the user is a single line of text as opposed to a paragraph.',
        recipientEmail: 'jenny@mail.com',

      },
      {
        id: '3',
        conversationId: '103',
        avatarImageURL: 'http://dummyimage.com/mediumrectangle/cccccc/eeeeee',
        recipientName: 'Mr. Jhonothan Goldsmith',
        lastMessage: 'yt',
        recipientEmail: 'Jhonothan@mail.com',
      },
      {
        id: '4',
        conversationId: '105',
        avatarImageURL: 'http://dummyimage.com/mediumrectangle/222222/eeeeee',
        recipientName: 'Ron',
        lastMessage: 'hola',
        recipientEmail: 'Ron@mail.com',
      },
      {
        id: '5',
        conversationId: '106',
        avatarImageURL: 'http://dummyimage.com/mediumrectangle/bbbbbbb/eeeeee',
        recipientName: 'Jimmy',
        lastMessage: ' A single line field is used when the input ',
        recipientEmail: 'Jimmy@mail.com',
      },
      {
        id: '6',
        conversationId: '107',
        avatarImageURL: 'http://dummyimage.com/mediumrectangle/333333/eeeeee',
        recipientName: 'Page',
        lastMessage: 'How is everything',
        recipientEmail: 'Page@mail.com',
      },
      {
        id: '7',
        conversationId: '105',
        avatarImageURL: 'http://dummyimage.com/mediumrectangle/dddddd/eeeeee',
        recipientName: 'Ash',
        lastMessage: 'Given the current situation',
        recipientEmail: 'Ash@mail.com',
      },
      {
        id: '8',
        conversationId: '106',
        avatarImageURL: 'http://dummyimage.com/mediumrectangle/999999/eeeeee',
        recipientName: 'Danny',
        lastMessage: 'And that was all',
        recipientEmail: 'Danny@mail.com',
      },
    ],
  },
};

export default friendListPageData;
