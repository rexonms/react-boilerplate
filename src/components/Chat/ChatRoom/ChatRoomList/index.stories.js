import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { withInfo } from '@storybook/addon-info';

import ChatRoomList from './index';
import { conversationList } from './dummyData';

const stories = storiesOf('components/Chat/ChatRoom/ChatRoomList', module);

stories.addDecorator(withKnobs);
stories.addDecorator(story => (story()));
stories.add('ChatRoomList', withInfo(`
  Component
  ~~~js
    <ChatRoomList name='ChatRoomList' />)
  ~~~
`)(() => {
  return (
    <ChatRoomList
      list={conversationList}
    />
  );
}));
