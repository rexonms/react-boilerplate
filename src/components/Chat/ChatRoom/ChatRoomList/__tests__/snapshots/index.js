import React from 'react';
import { shallow } from 'enzyme';

import ChatRoomList from '../../index';
import { conversationList } from '../../dummyData';
// https://hackernoon.com/oh-snap-snapshots-unit-integration-testing-with-jest-enzyme-13cc18aecb7b

it('Renders component correctly', () => {
  const component = shallow(<ChatRoomList list={conversationList} />);
  expect(component).toMatchSnapshot();
});
