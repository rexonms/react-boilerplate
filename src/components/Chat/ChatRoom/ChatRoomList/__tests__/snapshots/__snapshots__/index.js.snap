// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatRoomList
    list={
        Object {
            "1": Object {
              "authorEmail": "rexonms@gmail.com",
              "id": "1",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "hola!",
            },
            "2": Object {
              "authorEmail": "yi@gmail.com",
              "id": "2",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "whats up!",
            },
            "3": Object {
              "authorEmail": "yi@gmail.com",
              "id": "3",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "Where did you go? Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: ",
            },
            "4": Object {
              "authorEmail": "rexonms@gmail.com",
              "id": "4",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "Oops sorry! didn't se you message",
            },
            "5": Object {
              "authorEmail": "rexonms@gmail.com",
              "id": "5",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "In addition, you should probably review the links in the 'See Also' section of that document. I especially recommend the video presentations.
        
        Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: ",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="content user"
>
          <ChatRoomItem
                    id="1"
                    imageAltText="Alt text of image"
                    imageURL="https://place-hold.it/100"
                    message="hola!"
                    type="user"
          />
</div>,
        <div
          className="content friend"
>
          <ChatRoomItem
                    id="2"
                    imageAltText="Alt text of image"
                    imageURL="https://place-hold.it/100"
                    message="whats up!"
                    type="friend"
          />
</div>,
        <div
          className="content friend"
>
          <ChatRoomItem
                    id="3"
                    imageAltText="Alt text of image"
                    imageURL="https://place-hold.it/100"
                    message="Where did you go? Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: "
                    type="friend"
          />
</div>,
        <div
          className="content user"
>
          <ChatRoomItem
                    id="4"
                    imageAltText="Alt text of image"
                    imageURL="https://place-hold.it/100"
                    message="Oops sorry! didn't se you message"
                    type="user"
          />
</div>,
        <div
          className="content user"
>
          <ChatRoomItem
                    id="5"
                    imageAltText="Alt text of image"
                    imageURL="https://place-hold.it/100"
                    message="In addition, you should probably review the links in the 'See Also' section of that document. I especially recommend the video presentations.
          
          Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: "
                    type="user"
          />
</div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "conv-1",
        "nodeType": "host",
        "props": Object {
          "children": <ChatRoomItem
            id="1"
            imageAltText="Alt text of image"
            imageURL="https://place-hold.it/100"
            message="hola!"
            type="user"
/>,
          "className": "content user",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "1",
            "imageAltText": "Alt text of image",
            "imageURL": "https://place-hold.it/100",
            "message": "hola!",
            "type": "user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "conv-2",
        "nodeType": "host",
        "props": Object {
          "children": <ChatRoomItem
            id="2"
            imageAltText="Alt text of image"
            imageURL="https://place-hold.it/100"
            message="whats up!"
            type="friend"
/>,
          "className": "content friend",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "2",
            "imageAltText": "Alt text of image",
            "imageURL": "https://place-hold.it/100",
            "message": "whats up!",
            "type": "friend",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "conv-3",
        "nodeType": "host",
        "props": Object {
          "children": <ChatRoomItem
            id="3"
            imageAltText="Alt text of image"
            imageURL="https://place-hold.it/100"
            message="Where did you go? Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: "
            type="friend"
/>,
          "className": "content friend",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "3",
            "imageAltText": "Alt text of image",
            "imageURL": "https://place-hold.it/100",
            "message": "Where did you go? Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: ",
            "type": "friend",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "conv-4",
        "nodeType": "host",
        "props": Object {
          "children": <ChatRoomItem
            id="4"
            imageAltText="Alt text of image"
            imageURL="https://place-hold.it/100"
            message="Oops sorry! didn't se you message"
            type="user"
/>,
          "className": "content user",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "4",
            "imageAltText": "Alt text of image",
            "imageURL": "https://place-hold.it/100",
            "message": "Oops sorry! didn't se you message",
            "type": "user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "conv-5",
        "nodeType": "host",
        "props": Object {
          "children": <ChatRoomItem
            id="5"
            imageAltText="Alt text of image"
            imageURL="https://place-hold.it/100"
            message="In addition, you should probably review the links in the 'See Also' section of that document. I especially recommend the video presentations.

Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: "
            type="user"
/>,
          "className": "content user",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "5",
            "imageAltText": "Alt text of image",
            "imageURL": "https://place-hold.it/100",
            "message": "In addition, you should probably review the links in the 'See Also' section of that document. I especially recommend the video presentations.

Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: ",
            "type": "user",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="content user"
>
            <ChatRoomItem
                        id="1"
                        imageAltText="Alt text of image"
                        imageURL="https://place-hold.it/100"
                        message="hola!"
                        type="user"
            />
</div>,
          <div
            className="content friend"
>
            <ChatRoomItem
                        id="2"
                        imageAltText="Alt text of image"
                        imageURL="https://place-hold.it/100"
                        message="whats up!"
                        type="friend"
            />
</div>,
          <div
            className="content friend"
>
            <ChatRoomItem
                        id="3"
                        imageAltText="Alt text of image"
                        imageURL="https://place-hold.it/100"
                        message="Where did you go? Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: "
                        type="friend"
            />
</div>,
          <div
            className="content user"
>
            <ChatRoomItem
                        id="4"
                        imageAltText="Alt text of image"
                        imageURL="https://place-hold.it/100"
                        message="Oops sorry! didn't se you message"
                        type="user"
            />
</div>,
          <div
            className="content user"
>
            <ChatRoomItem
                        id="5"
                        imageAltText="Alt text of image"
                        imageURL="https://place-hold.it/100"
                        message="In addition, you should probably review the links in the 'See Also' section of that document. I especially recommend the video presentations.
            
            Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: "
                        type="user"
            />
</div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "conv-1",
          "nodeType": "host",
          "props": Object {
            "children": <ChatRoomItem
              id="1"
              imageAltText="Alt text of image"
              imageURL="https://place-hold.it/100"
              message="hola!"
              type="user"
/>,
            "className": "content user",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "1",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "hola!",
              "type": "user",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "conv-2",
          "nodeType": "host",
          "props": Object {
            "children": <ChatRoomItem
              id="2"
              imageAltText="Alt text of image"
              imageURL="https://place-hold.it/100"
              message="whats up!"
              type="friend"
/>,
            "className": "content friend",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "2",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "whats up!",
              "type": "friend",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "conv-3",
          "nodeType": "host",
          "props": Object {
            "children": <ChatRoomItem
              id="3"
              imageAltText="Alt text of image"
              imageURL="https://place-hold.it/100"
              message="Where did you go? Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: "
              type="friend"
/>,
            "className": "content friend",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "3",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "Where did you go? Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: ",
              "type": "friend",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "conv-4",
          "nodeType": "host",
          "props": Object {
            "children": <ChatRoomItem
              id="4"
              imageAltText="Alt text of image"
              imageURL="https://place-hold.it/100"
              message="Oops sorry! didn't se you message"
              type="user"
/>,
            "className": "content user",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "4",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "Oops sorry! didn't se you message",
              "type": "user",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "conv-5",
          "nodeType": "host",
          "props": Object {
            "children": <ChatRoomItem
              id="5"
              imageAltText="Alt text of image"
              imageURL="https://place-hold.it/100"
              message="In addition, you should probably review the links in the 'See Also' section of that document. I especially recommend the video presentations.

Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: "
              type="user"
/>,
            "className": "content user",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "5",
              "imageAltText": "Alt text of image",
              "imageURL": "https://place-hold.it/100",
              "message": "In addition, you should probably review the links in the 'See Also' section of that document. I especially recommend the video presentations.

Finally, you should probably take a look at this document for a discussion of the three possible schemas for a messaging/commenting database, including the trade-offs for each design: ",
              "type": "user",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
